examples/xlio_ultra_api_ping_pong.c

```diff
@@ -972,9 +832,451 @@ static void process_received_message(xlio_socket_t sock, const char *msg, size_t
+
+    /* Reset global variables for clean parsing */
+    optind = 1;
+
+    while ((option = getopt_long(argc, argv, "sci:p:I:n:t:h", long_options, &option_index)) != -1) {
+        switch (option) {
+        case 's':
+            if (mode_specified) {
+                fprintf(stderr, "Error: Cannot specify both server and client modes\n");
+                return -1;
+            }
+            g_config.is_server = true;
+            mode_specified = true;
+            break;
+
+        case 'c':
+            if (mode_specified) {
+                fprintf(stderr, "Error: Cannot specify both server and client modes\n");
+                return -1;
+            }
+            g_config.is_server = false;
+            mode_specified = true;
+            break;
+
+        case 'i':
+            g_config.ip = optarg;
+            break;
+
+        case 'p':
+            int port_val = atoi(optarg);
+            if (port_val <= 0 || port_val > 65535) {
+                fprintf(stderr, "Error: Invalid port number: %s\n", optarg);
+                return -1;
+            }
+            g_config.port = (unsigned short)port_val;
+            break;
+
+        case 'I':
+            g_config.client_ip = optarg;
+            break;
+
+        case 'n':
+            g_config.ping_count = atoi(optarg);
+            if (g_config.ping_count <= 0) {
+                fprintf(stderr, "Error: Invalid ping count: %s\n", optarg);
+                return -1;
+            }
+            break;
+
+        case 't':
+            g_config.timeout_seconds = atoi(optarg);
+            if (g_config.timeout_seconds < 0) {
+                fprintf(stderr, "Error: Timeout must be non-negative (0 = no timeout)\n");
+                return -1;
+            }
+            break;
+
+        case 'h':
+            print_usage(argv[0]);
+            return -1;
+
+        case '?':
+            /* getopt_long already printed error message */
+            fprintf(stderr, "Use --help for usage information.\n");
+            return -1;
+
+        default:
+            fprintf(stderr, "Error: Unknown option\n");
+            return -1;
+        }
+    }
+
+    /* Check for extra arguments */
+    if (optind < argc) {
+        fprintf(stderr, "Error: Unexpected arguments:");
+        while (optind < argc) {
+            fprintf(stderr, " %s", argv[optind++]);
+        }
+        fprintf(stderr, "\nUse --help for usage information.\n");
+        return -1;
+    }
+
+    /* Validate that mode was specified */
+    if (!mode_specified) {
+        fprintf(stderr, "Error: Must specify either --server (-s) or --client (-c) mode\n");
+        print_usage(argv[0]);
+        return -1;
+    }
+
+    /* Client mode validation */
+    if (!g_config.is_server) {
+        if (!g_config.ip || g_config.ip[0] == '\0') {
+            fprintf(stderr, "Error: Client mode requires server IP address (--ip)\n");
+            return -1;
+        }
+    }
+
+    /*
+     * Setup Timeout (if specified)
```

Comment:
consider enhacning:
`    /*
     * Setup Timeout (if specified)
     * Useful for CI/CD pipelines and automated testing where we need
     * the example to exit after a certain time even if no events occur.
     */`